zap/src/main/java/org/zaproxy/zap/extension/custompages/DefaultCustomPage.java

```diff
@@ -0,0 +1,279 @@
+     * @param regex a boolean specifying whether or not the {@code DefaultCustomPage} page matcher
+     *     is represented by a Regex pattern
+     * @param type the {@code CustomPage.Type} of the {@code DefaultCustomPage}
+     * @param enabled a boolean specifying whether or not the {@code DefaultCustomPage} is enabled
+     *     or not
+     */
+    public DefaultCustomPage(
+            int contextId,
+            String pageMatcher,
+            int pageMatcherTypeID,
+            boolean regex,
+            Type type,
+            boolean enabled) {
+        super();
+        this.contextId = contextId;
+        this.pageMatcher = pageMatcher;
+        this.pageMatcherType =
+                CustomPageMatcherLocation.getCustomPagePageMatcherTypeWithId(pageMatcherTypeID);
+        this.regex = regex;
+        this.type = type;
+        this.setEnabled(enabled);
+    }
+
+    @Override
+    public int getContextId() {
+        return contextId;
+    }
+
+    @Override
+    public void setContextId(int contextId) {
+        this.contextId = contextId;
+    }
+
+    @Override
+    public String getPageMatcher() {
+        return pageMatcher;
+    }
+
+    @Override
+    public void setPageMatcher(String pageMatcher) {
+        this.pageMatcher = pageMatcher;
+    }
+
+    @Override
+    public CustomPageMatcherLocation getPageMatcherType() {
+        return pageMatcherType;
+    }
+
+    @Override
+    public void setPageMatcherType(CustomPageMatcherLocation cppmt) {
+        this.pageMatcherType = cppmt;
+    }
+
+    @Override
+    public boolean isRegex() {
+        return regex;
+    }
+
+    @Override
+    public void setRegex(boolean regex) {
+        this.regex = regex;
+    }
+
+    @Override
+    public Type getType() {
+        return type;
+    }
+
+    @Override
+    public void setType(Type cpt) {
+        this.type = cpt;
+    }
+
+    /**
+     * Determines if a {@code HttpMessage} is a Custom Page of a particular {@code CustomPageType}.
+     *
+     * @param msg the HTTP message to be evaluated
+     * @param cpt the CustomPageType of the Custom Pages against which the HTTP message should be
+     *     evaluated
+     * @return {@code true} if the HTTP message is a Custom Page of the type in question, {@code
+     *     false} otherwise
+     */
+    public boolean isCustomPage(HttpMessage msg, Type cpt) {
+        if (isEnabled() && getType() == cpt) {
+            String value = getPageMatcherByType(msg);
+            return matchByType(value);
+        }
+        return false; // Default
+    }
+
+    private String getPageMatcherByType(HttpMessage msg) {
+        if (CustomPageMatcherLocation.URL.equals(getPageMatcherType())) {
+            return msg.getRequestHeader().getURI().toString();
+        } else if (CustomPageMatcherLocation.RESPONSE_CONTENT.equals(getPageMatcherType())) {
+            return getHttpMessageAsString(msg);
+        }
+        LOGGER.error("Could not get page matcher for the given message.");
```

Comment:
`... message, with: " + getPageMatcherType()` ? (The getter and variable should be renamed to use "location".)